File: MovieFinder.Core/Dtos/MovieDto.cs
namespace MovieFinder.Core.Dtos;

public class MovieDto
{
    public string Id { get; set; }
    public string Name { get; set; }
    public int Year { get; set; }
    public int RunTime { get; set; }
    public DateTime ReleaseDate { get; set; }
    public string? StoryLine { get; set; }
    public IEnumerable<CategoryDto> Categories { get; set; }
    public IEnumerable<NameDto> Directors { get; set; }
    public IEnumerable<NameDto> Writers { get; set; }
    public IEnumerable<NameDto> Actors { get; set; }
}

File: MovieFinder.Web/ClientApp/src/core/models/movieDto.ts
import {CategoryDto} from "./categoryDto";
import {NameDto} from "./nameDto";

export interface MovieDto {
    id: string;
    name: string;
    year: number;
    runTime: number;
    releaseDate: Date;
    storyLine: string;
    categories: CategoryDto[];
    directors: NameDto[];
    writers: NameDto[];
    actors: NameDto[];
}